File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/postgresql/PgDomainTypeScalar.scala
package dev.typr.foundationssc.docs.postgresql
import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object PgDomainTypeScalar:
  // start
  // PG schema:  CREATE DOMAIN person_name AS varchar(100);
  case class Name(value: String)
  object Name:
    val pgType: PgType[Name] =
      PgTypes.text.transform(Name.apply, _.value).asDomain("person_name")
  // stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/postgresql/PgDomainTypeScalar.java
package dev.typr.foundations.docs.postgresql;

import dev.typr.foundations.PgType;
import dev.typr.foundations.PgTypes;

@SuppressWarnings("unused")
public class PgDomainTypeScalar {
  // start
  // PG schema:  CREATE DOMAIN person_name AS varchar(100);
  public record Name(String value) {
    public static final PgType<Name> pgType =
        PgTypes.text.asDomain("person_name", Name::new, Name::value);
  }
  // stop
}
